FIx warnings and problems detected by Infer code analysis

Static code analysis from our CI ( `Static Code Analysis`) detected some warnings which should be fixed.

<details>
  <summary>Report</summary>
  

  ```js
#0
src/xml/joybuttonslotxml.cpp:129: error: Dead Store
  The value written to &firstTimePlus (type _Bool) is never used.
  127.                 extraStringData = QString();
  128.                 slotMixString = QString();
  129.                 firstTimePlus = true;
                       ^
  130. 
  131.                 xml->readNextStartElement();

#1
src/xml/inputdevicexml.cpp:249: error: Dead Store
  The value written to &direction (type int) is never used.
  247.                     {
  248.                         int type = xml->attributes().value("type").toString().toInt();
  249.                         int direction = xml->attributes().value("button").toString().toInt();
                               ^
  250.                         QString temp = xml->readElementText();
  251.                         if (!temp.isEmpty())

#2
src/xml/inputdevicexml.cpp:248: error: Dead Store
  The value written to &type (type int) is never used.
  246.                     } else if ((xml->name() == "sensorbuttonname") && xml->isStartElement())
  247.                     {
  248.                         int type = xml->attributes().value("type").toString().toInt();
                               ^
  249.                         int direction = xml->attributes().value("button").toString().toInt();
  250.                         QString temp = xml->readElementText();

#3
src/xml/inputdevicexml.cpp:298: error: Dead Store
  The value written to &type (type int) is never used.
  296.                     } else if ((xml->name() == "sensorname") && xml->isStartElement())
  297.                     {
  298.                         int type = xml->attributes().value("type").toString().toInt();
                               ^
  299.                         QString temp = xml->readElementText();
  300.                         if (!temp.isEmpty())

#4
src/mousedialog/springmoderegionpreview.cpp:74: error: Dead Store
  The value written to &tempwidth (type int) is never used.
  72. int SpringModeRegionPreview::adjustSpringSizeWidth(int width)
  73. {
  74.     int tempwidth = size().width();
          ^
  75.     if (width >= 2)
  76.     {

#5
src/mousedialog/springmoderegionpreview.cpp:88: error: Dead Store
  The value written to &tempheight (type int) is never used.
  86. int SpringModeRegionPreview::adjustSpringSizeHeight(int height)
  87. {
  88.     int tempheight = size().height();
          ^
  89. 
  90.     if (height >= 2)

#6
src/eventhandlers/uinputeventhandler.cpp:197: error: Dead Store
  The value written to &tempcode (type unsigned int) is never used.
  195.         if (code <= 3)
  196.         {
  197.             unsigned int tempcode = BTN_LEFT;
                   ^
  198.             switch (code)
  199.             {

#7
src/joydpad.cpp:319: error: Dead Store
  The value written to &initialCurve (type int) is never used.
  317.     bool result = true;
  318. 
  319.     JoyButton::JoyMouseCurve initialCurve = JoyButton::LinearCurve;
           ^
  320. 
  321.     QHash<int, JoyDPadButton *> temphash = getApplicableButtons();

#8
src/x11extras.cpp:559: error: Null Dereference
  pointer `null_char` last assigned on line 556 could be null and is dereferenced at line 559, column 13.
  557.         if (((reinterpret_cast<char *>(prop)) + nitems - 1) > null_char)
  558.         {
  559.             *(null_char) = ' ';
                   ^
  560.         }
  561. 

#9
src/inputdevice.cpp:606: error: Null Dereference
  pointer `button` last assigned on line 595 could be null and is dereferenced at line 606, column 5.
  604. 
  605.     JoyButton::SetChangeCondition tempmode = static_cast<JoyButton::SetChangeCondition>(mode);
  606.     button->setChangeSetSelection(originset);
           ^
  607.     button->setChangeSetCondition(tempmode, true);
  608. }

#10
src/x11extras.cpp:657: error: Dead Store
  The value written to &feedbacks (type XFeedbackState*) is never used.
  655. 
  656.             XFree(feedbacks);
  657.             feedbacks = temp = nullptr;
                   ^
  658. 
  659.             if (feedback_id <= -1)

#11
src/x11extras.cpp:753: error: Dead Store
  The value written to &feedbacks (type XFeedbackState*) is never used.
  751. 
  752.             XFree(feedbacks);
  753.             feedbacks = temp = nullptr;
                   ^
  754.             XCloseDevice(display, device);
  755.         }

#12
src/gui/axiseditdialog.cpp:722: error: Uninitialized Value
  The value read from mode was never initialized.
  720.         break;
  721.     }
  722.     m_axis->setHapticTriggerMode(mode);
           ^
  723. }

#13
src/x11extras.cpp:773: error: Dead Store
  The value written to &mouse_device (type XIDeviceInfo*) is never used.
  771.         if ((current_devices->use == XISlavePointer) && (QString::fromUtf8(current_devices->name) == pointerName))
  772.         {
  773.             mouse_device = current_devices;
                   ^
  774.         }
  775.     }

#14
src/gui/advancebuttondialog.cpp:895: error: Dead Store
  The value written to &chosen_set (type int) is never used.
  893.     PadderCommon::inputDaemonMutex.lock();
  894. 
  895.     int chosen_set = -1;
           ^
  896.     JoyButton::SetChangeCondition set_selection_condition = JoyButton::SetChangeDisabled;
  897. 

#15
src/joybutton.cpp:1143: error: Dead Store
  The value written to &elapsedDiff (type double) is never used.
  1141.                             // Determine the multiplier to use for the current maximum mouse speed
  1142.                             // based on how much time has passed.
  1143.                             double elapsedDiff = 1.0;
                                    ^
  1144.                             if ((easingDuration > 0.0) && ((easingElapsed * .001) < easingDuration))
  1145.                             {

#16
src/joybutton.cpp:1294: error: Dead Store
  The value written to &elapsedDiff (type double) is never used.
  1292. 
  1293.                         double tempAccel = currentAccelMultiTemp;
  1294.                         double elapsedDiff = 1.0;
                                ^
  1295. 
  1296.                         if ((elapsedDuration > 0.0) && ((elapsedElapsed * 0.001) < elapsedDuration))

#17
src/joybutton.cpp:1308: error: Dead Store
  The value written to &elapsedDiff (type double) is never used.
  1306.                         } else
  1307.                         {
  1308.                             elapsedDiff = 1.0;
                                    ^
  1309.                             currentAccelMulti = 0.0;
  1310.                             updateOldAccelMulti = 0.0;

#18
src/joycontrolstick.cpp:1988: error: Dead Store
  The value written to &initialCurve (type int) is never used.
  1986.     bool result = true;
  1987. 
  1988.     JoyButton::JoyMouseCurve initialCurve = JoyButton::LinearCurve;
            ^
  1989.     QHash<JoyStickDirections, JoyControlStickButton *> temphash = getApplicableButtons();
  1990.     auto iter = temphash.cbegin();

#19
src/joybutton.cpp:3804: error: Dead Store
  The value written to &tempx (type double) is never used.
  3802.         for (int i = (springXSpeeds->length() - 1); (i >= 0) && !complete; i--)
  3803.         {
  3804.             double tempx = -2.0;
                    ^
  3805.             double tempy = -2.0;
  3806.             double tempSpringDeadX = 0.0;

#20
src/joybutton.cpp:3805: error: Dead Store
  The value written to &tempy (type double) is never used.
  3803.         {
  3804.             double tempx = -2.0;
  3805.             double tempy = -2.0;
                    ^
  3806.             double tempSpringDeadX = 0.0;
  3807.             double tempSpringDeadY = 0.0;

Found 21 issues
                Issue Type(ISSUED_TYPE_ID): #
                    Dead Store(DEAD_STORE): 18
        Null Dereference(NULL_DEREFERENCE): 2
  Uninitialized Value(UNINITIALIZED_VALUE): 1
  ```
</details>

Most of them are very simple ones, mostly never used variables.

--- a/src/eventhandlers/uinputeventhandler.cpp
+++ b/src/eventhandlers/uinputeventhandler.cpp
@@ -194,7 +194,7 @@ void UInputEventHandler::sendMouseButtonEvent(JoyButtonSlot *slot, bool pressed)
     {
         if (code <= 3)
         {
-            unsigned int tempcode = BTN_LEFT;
+            unsigned int tempcode;
             switch (code)
             {
             case 3: {

--- a/src/gui/advancebuttondialog.cpp
+++ b/src/gui/advancebuttondialog.cpp
@@ -892,7 +892,7 @@ void AdvanceButtonDialog::updateSetSelection()
 {
     PadderCommon::inputDaemonMutex.lock();
 
-    int chosen_set = -1;
+    int chosen_set;
     JoyButton::SetChangeCondition set_selection_condition = JoyButton::SetChangeDisabled;
 
     if (ui->setSelectionComboBox->currentIndex() > 0)

--- a/src/inputdevice.cpp
+++ b/src/inputdevice.cpp
@@ -600,11 +600,17 @@ void InputDevice::changeSetAxisButtonAssociation(int button_index, int axis_inde
     } else if (button_index == 1)
     {
         button = getJoystick_sets().value(newset)->getJoyAxis(axis_index)->getPAxisButton();
+    } else
+    {
+        WARN() << "Invalid button_index value: " << button_index;
+        return;
     }
-
-    JoyButton::SetChangeCondition tempmode = static_cast<JoyButton::SetChangeCondition>(mode);
-    button->setChangeSetSelection(originset);
-    button->setChangeSetCondition(tempmode, true);
+    if (button != nullptr)
+    {
+        JoyButton::SetChangeCondition tempmode = static_cast<JoyButton::SetChangeCondition>(mode);
+        button->setChangeSetSelection(originset);
+        button->setChangeSetCondition(tempmode, true);
+    }
 }
 
 void InputDevice::changeSetStickButtonAssociation(int button_index, int stick_index, int originset, int newset, int mode)

--- a/src/joybutton.cpp
+++ b/src/joybutton.cpp
@@ -1140,7 +1140,7 @@ void JoyButton::mouseEvent()
 
                             // Determine the multiplier to use for the current maximum mouse speed
                             // based on how much time has passed.
-                            double elapsedDiff = 1.0;
+                            double elapsedDiff;
                             if ((easingDuration > 0.0) && ((easingElapsed * .001) < easingDuration))
                             {
                                 elapsedDiff = ((easingElapsed * .001) / easingDuration);
@@ -1291,7 +1291,7 @@ void JoyButton::mouseEvent()
                         }
 
                         double tempAccel = currentAccelMultiTemp;
-                        double elapsedDiff = 1.0;
+                        double elapsedDiff;
 
                         if ((elapsedDuration > 0.0) && ((elapsedElapsed * 0.001) < elapsedDuration))
                         {
@@ -1305,7 +1305,6 @@ void JoyButton::mouseEvent()
                             updateOldAccelMulti = currentAccelMulti;
                         } else
                         {
-                            elapsedDiff = 1.0;
                             currentAccelMulti = 0.0;
                             updateOldAccelMulti = 0.0;
                             accelTravel = 0.0;
@@ -3801,8 +3800,6 @@ void JoyButton::moveSpringMouse(int &movedX, int &movedY, bool &hasMoved, int sp
 
         for (int i = (springXSpeeds->length() - 1); (i >= 0) && !complete; i--)
         {
-            double tempx = -2.0;
-            double tempy = -2.0;
             double tempSpringDeadX = 0.0;
             double tempSpringDeadY = 0.0;
 
@@ -3812,8 +3809,8 @@ void JoyButton::moveSpringMouse(int &movedX, int &movedY, bool &hasMoved, int sp
             infoX = springXSpeeds->takeLast();
             infoY = springYSpeeds->takeLast();
 
-            tempx = infoX.displacementX;
-            tempy = infoY.displacementY;
+            double tempx = infoX.displacementX;
+            double tempy = infoY.displacementY;
             tempSpringDeadX = infoX.springDeadX;
             tempSpringDeadY = infoY.springDeadY;
 

--- a/src/joycontrolstick.cpp
+++ b/src/joycontrolstick.cpp
@@ -1985,7 +1985,6 @@ bool JoyControlStick::hasSameButtonsMouseCurve()
 {
     bool result = true;
 
-    JoyButton::JoyMouseCurve initialCurve = JoyButton::LinearCurve;
     QHash<JoyStickDirections, JoyControlStickButton *> temphash = getApplicableButtons();
     auto iter = temphash.cbegin();
 
@@ -1993,7 +1992,7 @@ bool JoyControlStick::hasSameButtonsMouseCurve()
         return result;
 
     JoyControlStickButton *button = iter.value();
-    initialCurve = button->getMouseCurve();
+    JoyButton::JoyMouseCurve initialCurve = button->getMouseCurve();
 
     while (++iter != temphash.cend())
     {

--- a/src/joydpad.cpp
+++ b/src/joydpad.cpp
@@ -316,8 +316,6 @@ bool JoyDPad::hasSameButtonsMouseCurve()
 {
     bool result = true;
 
-    JoyButton::JoyMouseCurve initialCurve = JoyButton::LinearCurve;
-
     QHash<int, JoyDPadButton *> temphash = getApplicableButtons();
     auto iter = temphash.cbegin();
 
@@ -325,7 +323,7 @@ bool JoyDPad::hasSameButtonsMouseCurve()
         return result;
 
     JoyDPadButton *button = iter.value();
-    initialCurve = button->getMouseCurve();
+    JoyButton::JoyMouseCurve initialCurve = button->getMouseCurve();
 
     while (++iter != temphash.cend())
     {

--- a/src/mousedialog/springmoderegionpreview.cpp
+++ b/src/mousedialog/springmoderegionpreview.cpp
@@ -85,7 +85,7 @@ int SpringModeRegionPreview::adjustSpringSizeWidth(int width)
 
 int SpringModeRegionPreview::adjustSpringSizeHeight(int height)
 {
-    int tempheight = size().height();
+    int tempheight;
 
     if (height >= 2)
     {

--- a/src/x11extras.cpp
+++ b/src/x11extras.cpp
@@ -554,7 +554,7 @@ QString X11Extras::getWindowClass(Window window)
         qDebug() << nitems;
 
         char *null_char = strchr(reinterpret_cast<char *>(prop), '\0');
-        if (((reinterpret_cast<char *>(prop)) + nitems - 1) > null_char)
+        if ((null_char != nullptr) && (((reinterpret_cast<char *>(prop)) + nitems - 1) > null_char))
         {
             *(null_char) = ' ';
         }
@@ -654,7 +654,6 @@ void X11Extras::x11ResetMouseAccelerationChange(QString pointerName)
             checkFeedback(temp, num_feedbacks, feedback_id);
 
             XFree(feedbacks);
-            feedbacks = temp = nullptr;
 
             if (feedback_id <= -1)
             {
@@ -750,7 +749,6 @@ struct X11Extras::ptrInformation X11Extras::getPointInformation(QString pointerN
             }
 
             XFree(feedbacks);
-            feedbacks = temp = nullptr;
             XCloseDevice(display, device);
         }
 

--- a/src/xml/joybuttonslotxml.cpp
+++ b/src/xml/joybuttonslotxml.cpp
@@ -126,7 +126,6 @@ void JoyButtonSlotXml::readConfig(QXmlStreamReader *xml)
                 tempStringData = QString();
                 extraStringData = QString();
                 slotMixString = QString();
-                firstTimePlus = true;
 
                 xml->readNextStartElement();
             }

